### PR TITLE
fix: gateway migration et al.

### PR DESCRIPTION
**Pull request by wischli**
# Description

* Fixes Gateway migration from https://github.com/centrifuge/centrifuge-chain/pull/1946
* Removes DMP Queue from runtimes because this was deprecated with Polkadot v1.7.2 (or lower) but we needed to keep the pallet in order to migrate to the new Polkadot SDK Message Queue. This is done now and DMP Queue can be removed safely.
* Removes deprecated common migrations for loans v4 and hold reasons from Polkadot v1.7.2
* Adds migration to remove undecodable `ForeignInvestmentInfo` v1 storage entries (see [Slack for context](https://kflabs.slack.com/archives/C04GNHLFB6Z/p1722964161013639))

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by cdamian** [comment]
Thank you very much @wischli!!! <3

**Review by cdamian — APPROVED** [review]

File: runtime/common/src/migrations/foreign_investments_v2.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+const LOG_PREFIX: &str = "LiquidityPoolsGatewayV1";
+
+impl<T> OnRuntimeUpgrade for Migration<T>
+where
+	T: frame_system::Config,
+{
+	fn on_runtime_upgrade() -> Weight {
+		let mut weight = Weight::zero();
+
+		match clear_prefix(&get_storage_prefix(), Some(REMOVAL_LIMIT)) {
```
[the remark this answers is not in the document]
Comment (lemunozm) [reply runtime/common/src/migrations/foreign_investments_v2.rs:30]: Agree! Thanks for the explanation

**Review by lemunozm — APPROVED** [review]
Thanks @wischli for this clean up and for being to much aware of all details of migrations

**Comment by lemunozm** [comment]
It seems like our try-runtime CI job is broken 🤔

**Comment by wischli** [comment]
> It seems like our try-runtime CI job is broken 🤔

I hope [9dc9c03](https://github.com/centrifuge/centrifuge-chain/pull/1950/commits/9dc9c034dbd229638f193f919b55cf665397d501) will fix it.

**Pull request merged**